src/forms/calendarComponent.test.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import CalendarComponent from './CalendarComponent';
+
+describe('CalendarComponent', function() {
+
+	let calendarComponent;
+
+	const id = 'beyonce',
+		name = 'halo',
+		className = 'beyonce-halo',
+		value = new Date('2012-10-12'),
+		newValue = new Date('2020-10-12'),
+		opts = { dateFormat: 'm-d-Y' };
+
+	const onChangeCallback = jest.fn();
+
+	beforeEach(() => {
+		calendarComponent = shallow(
+			<CalendarComponent
+				id={id}
+				name={name}
+				className={className}
+				value={value}
+				datepickerOptions={opts}
+				onChangeCallback={onChangeCallback}
+				ref={ node => this.node = node }
+			/>
+		);
+		calendarComponent.instance().componentDidMount();
+	});
+
+	afterEach(() => {
+		calendarComponent = null;
+	});
+
+	it('renders the calendarComponent with appropriate attributes', function() {
+		expect(calendarComponent).toMatchSnapshot();
+	});
+
+	describe('onChange callback tests', () => {
+
+		let calendarComponent,
+			onChangePropMock,
+			callbackMock;
+
+		beforeEach(() => {
+			onChangePropMock = jest.fn();
+			callbackMock = jest.fn();
+
+			calendarComponent = shallow(
+				<CalendarComponent
+					id={id}
+					name={name}
+					className={className}
+					onChange={onChangePropMock}
+					onChangeCallback={callbackMock}
+				/>
+			);
+		});
+
+		afterEach(() => {
+			calendarComponent = null;
+		});
+
+		// because we are rendering shallow here, flatpickr isnt
+		// instantiated (because componentDidMount not called)
+		// just using direct calls to flatpickr change to test
+
+		it('calls onChange prop if one is provided, as with redux-form', function() {
```

Comment:
since you only have one `it()` test, I recommend _not_ using `beforeEach`/`afterEach` - they are generally recommended only when they save you a lot of setup/teardown code for many different unit tests, so in this case it would be better to put everything inside the `it()` - then you won't have to deal with the complex parent scope issues.

Reply:
👍